src/main/java/org/folio/rest/helper/FundsHelper.java

```diff
@@ -50,8 +64,50 @@ public FundsHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
     return handleDeleteRequest(resourceByIdPath(FUND_TYPES, id, lang));
   }
 
-  public CompletableFuture<Fund> createFund(Fund fund) {
-    return handleCreateRequest(resourcesPath(FUNDS), fund).thenApply(fund::withId);
+  public CompletableFuture<CompositeFund> createFund(CompositeFund compositeFund) {
+    final Fund fund = compositeFund.getFund();
+    if (CollectionUtils.isNotEmpty(compositeFund.getGroupIds())) {
+      return getCurrentFiscalYearId(fund.getLedgerId())
+        .thenCompose(fiscalYearId -> handleCreateRequest(resourcesPath(FUNDS), fund).thenAccept(fund::setId)
+          .thenCompose(ok -> assignFundToGroups(compositeFund, fiscalYearId)))
+        .thenApply(aVoid -> compositeFund);
+    }
+    return handleCreateRequest(resourcesPath(FUNDS), fund)
+      .thenApply(ok -> compositeFund);
+  }
+
+  private CompletableFuture<String> getCurrentFiscalYearId(String ledgerId) {
+
+    return new LedgersHelper(httpClient, okapiHeaders, ctx, lang).getLedger(ledgerId)
+      .thenCompose(ledger -> new FiscalYearsHelper(httpClient, okapiHeaders, ctx, lang).getFiscalYear(ledger.getFiscalYearOneId()))
+      .thenApply(this::buildCurrentFYEndpoint)
+      .thenCompose(endpoint -> handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
+        .thenApply(fiscalYears -> fiscalYears.mapTo(FiscalYearsCollection.class))
+        .thenApply(fiscalYearsCollection -> fiscalYearsCollection.getFiscalYears().stream()
+          .map(FiscalYear::getId).findFirst()
+          .orElseThrow(() -> new HttpException(422, FISCAL_YEARS_NOT_FOUND)))
+      );
+
+  }
+
+  private String buildCurrentFYEndpoint(FiscalYear fiscalYearOne) {
+    Instant now = Instant.now();
+    Instant next = now.plus(HelperUtils.getFiscalYearDuration(fiscalYearOne));
+    String query = String.format(SEARCH_CURRENT_FISCAL_YEAR_QUERY, fiscalYearOne.getSeries(), now, now, next, next);
+    return String.format(GET_FISCAL_YEARS_BY_QUERY, 1, 0, buildQueryParam(query, logger), lang);
+  }
+
+  private CompletableFuture<Void> assignFundToGroups(CompositeFund compositeFund, String fiscalYearId) {
+    List<GroupFundFiscalYear> groupFundFiscalYears = buildGroupFundFiscalYear(compositeFund, fiscalYearId);
+    return VertxCompletableFuture.allOf(ctx, groupFundFiscalYears.stream()
+      .map(groupFundFiscalYear -> new GroupFundFiscalYearHelper(httpClient, okapiHeaders, ctx, lang).createGroupFundFiscalYear(groupFundFiscalYear))
```

Comment:
what if one of the groundFundFY record fails to create? Probably the entire request will fail , but there will be some records already created?

Reply:
As this is merged. Can you create a small story so that we don't lose track, Thanks!